src/ngAria/aria.js

```diff
@@ -377,6 +377,25 @@ ngAriaModule.directive('ngShow', ['$aria', function($aria) {
             });
           }
         }
+
+        if (elem[0].nodeName === 'A') {
+          if ($aria.config('bindRoleForClick') && !attr.href && !attr.xlinkHref && !attr.role) {
+              elem.attr('role', 'link');
+          }
+          if ($aria.config('bindKeypress')) {
```

Comment:
To avoid unnecessary work, we should check for `!attr.ngKeypress` here (instead of inside the listener below).